modin/pandas/base.py

```diff
@@ -3426,6 +3426,80 @@ def __and__(self, other):
     def __rand__(self, other):
         return self._binary_op("__rand__", other, axis=0)
 
+    def __array_function__(self, func, types, args, kwargs):
+        """
+        Return the result of calling an array function on self.
+
+        Parameters
+        ----------
+        func : Callable
+            The function to call.
+        types : list[types]
+            Types of arguments.
+        args : list
+            Arguments to pass to function.
+        kwargs : dict
+            Key word arguments to pass to function.
+
+        Returns
+        -------
+        arr : np.ndarray or modin.numpy.array
+            The result of calling the array function on self.
+        """
+        out = self.to_numpy().__array_function__(func, types, args, kwargs)
+        if out is NotImplemented:
+            func_name = func.__name__
+            arr = self.__array__()
+            if ExperimentalNumPyAPI.get():
+                ErrorMessage.warn(
+                    f"Attempted to use Experimental NumPy API for function {func_name} but failed. Defaulting to NumPy."
+                )
+            converted_args = []
+            for input in args:
+                if hasattr(input, "_query_compiler"):
+                    input = input.__array__()
+                converted_args += [input]
+            where_kwarg = kwargs.get("where")
+            if where_kwarg is not None:
+                if hasattr(where_kwarg, "_query_compiler"):
+                    kwargs["where"] = where_kwarg.__array__()
+            return func(arr, *converted_args[1:], **kwargs)
```

Comment:
this seems to be losing `converteg_args[0]` (which was `args[0]`), feels weird

Reply:
I can add a comment - but basically args[0] is self. Since we’re calling the function directly instead of calling the special method again, I’m passing in the converted self as the first argument, and only need to pass in any remaining args (after making sure to convert them)

Reply:
a comment would be nice; also, if we're dropping `converted_args[0]`, then we should probably be doing so in the upper loop and skip the call of `args[0].__array__()` altogether